src/app/clusters/camera-av-settings-user-level-management-server/camera-av-settings-user-level-management-server.cpp

```diff
@@ -0,0 +1,1079 @@
+            return;
+        }
+        uint8_t zoomValue = zoom.Value();
+        if ((zoomValue > mZoomMax) || (zoomValue < kMinZoomValue))
+        {
+            ChipLogError(Zcl, "Received Zoom value out of range");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+        hasAtLeastOneValue = true;
+    }
+
+    // Was a value received in the command
+    if (!hasAtLeastOneValue)
+    {
+        ChipLogError(Zcl, "MPTZSetPosition contains no actionable fields");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::InvalidCommand);
+        return;
+    }
+
+    // Check with the delegate that we're in a position to change any of the PTZ values
+    //
+    if (!mDelegate->CanChangeMPTZ())
+    {
+        ChipLogDetail(Zcl, "Device not able to process MPTZ change");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Busy);
+        return;
+    }
+
+    // Call the delegate to set the new values
+    //
+    status = mDelegate->MPTZSetPosition(pan, tilt, zoom);
+
+    if (status != Status::Success)
+    {
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Set the local values of pan, tilt, and zoom
+    SetPan(pan);
+    SetTilt(tilt);
+    SetZoom(zoom);
+
+    MarkDirty(Attributes::MPTZPosition::Id);
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, status);
+}
+
+void CameraAvSettingsUserLevelMgmtServer::HandleMPTZRelativeMove(HandlerContext & ctx,
+                                                                 const Commands::MPTZRelativeMove::DecodableType & commandData)
+{
+
+    Status status           = Status::Success;
+    bool hasAtLeastOneValue = false;
+
+    Optional<int16_t> panDelta  = commandData.panDelta;
+    Optional<int16_t> tiltDelta = commandData.tiltDelta;
+    Optional<int8_t> zoomDelta  = commandData.zoomDelta;
+    int16_t newPan              = mMptzPosition.pan.Value();
+    int16_t newTilt             = mMptzPosition.tilt.Value();
+    int8_t newZoom              = static_cast<int8_t>(mMptzPosition.zoom.Value());
+
+    // Validate the received command fields
+    //
+    if (panDelta.HasValue())
+    {
+        if (!HasFeature(Feature::kMechanicalPan))
+        {
+            ChipLogError(Zcl, "Mechanical Pan not supported although panDelta value provided");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+            return;
+        }
+        int16_t panDeltaValue = panDelta.Value();
+        if (panDeltaValue > (mPanMax - mPanMin) || panDeltaValue < -(mPanMax - mPanMin))
+        {
+            ChipLogError(Zcl, "PanDelta value received is out of range.");
+            ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+            return;
+        }
+
+        newPan = static_cast<int16_t>(newPan + panDeltaValue);
+        if (newPan > mPanMax)
```

Comment:
OK, so this is safe because the pan values are actually constrained so that panMax - panMin <= 360, right?  That might be worth a comment explaining why overflow can't happen here: `newPan` is between -180 and 180, and the thing being added is -360 to 360, so the sum can't overflow `int16_t`.

Reply:
Added comment.